pkg/rules/engine/engine.go

```diff
@@ -221,23 +185,47 @@ func (engine *Engine) dispatchEvent(s detect.Signature, event protocol.Event) {
 	engine.signatures[s] <- event
 }
 
-//LoadSignature will store in Engine data structures the given signature and activate its handling business logics.
+//LoadSignature will call the internal signature loading logic and activate its handling business logics.
 // It will return the signature ID as well as error.
 func (engine *Engine) LoadSignature(signature detect.Signature) (string, error) {
+	id, err := engine.loadSignature(signature)
+	if err != nil {
+		return id, err
+	}
+	engine.signaturesMutex.RLock()
+	go signatureStart(signature, engine.signatures[signature], &engine.waitGroup)
+	engine.signaturesMutex.RUnlock()
+
+	return id, nil
+}
+
+//loadSignature handles storing a signature in the Engine data structures
+// It will return the signature ID as well as error.
```

Comment:
I assume we don't return signature od and error at the same time.
```suggestion
// It will return the signature ID or error.
```